### PR TITLE
Improve MSBuild codegen handling of gbc changes

### PR DESCRIPTION
**Pull request by chwarr**
Two commits here:

* teach the MSBuild targets to rebuild when gbc itself is changed
* improve MSBuild targets comments

Please rebase when merging.

### CODE REVIEW

File: cs/build/nuget/Common.targets [hunk 1]
```diff
@@ -34,22 +34,44 @@
...
-    *    must rebuild all .bond files when any of them change.  We force that by putting a non-transform output
-    *    (bondcodegen.done) in the list so IFF any input changes, then all inputs rebuild since MSBuild can't
-    *    know what should create the unmapped output item and so can't be selective.
-    *    NOTE - this still won't catch changes to files outside declared BondCodegen elements.
-  -->
+  <!-- Attempt to find gbc in a platform-agnostic way. -->
+  <PropertyGroup>
+    <_BondExe>gbc</_BondExe>
+    <_BondExe Condition="Exists('$(BOND_COMPILER_PATH)\gbc.exe')">$(BOND_COMPILER_PATH)\gbc.exe</_BondExe>
+    <_BondExe Condition="Exists('$(BOND_COMPILER_PATH)\gbc')">$(BOND_COMPILER_PATH)\gbc</_BondExe>
```
Comment (tstein) [inline cs/build/nuget/Common.targets]: Just for the hell of it, what happens if both of these files exist? I haven't done this yet, but I think I could plausibly cause that situation if I shared a tree between a Windows-world build and and WSL-world one.

Comment (chwarr) [inline cs/build/nuget/Common.targets]: It would use the second one. Properties are evaluated top-to-bottom, and the last one that evaluates wins.

**Comment by chwarr** [comment]
CI failure is due to https://github.com/Microsoft/bond/issues/508, which @lalo is working on "fixing" with #512.

**Comment by chwarr** [comment]
Merged in commit 0bce7292.

**Pull request closed without merging**